### PR TITLE
Move the tests out to their own tasks

### PR DESCRIPTION
**Pull request by aaomidi**
https://github.com/microsoft/azuredatastudio/issues/11168

### CODE REVIEW

File: build/azure-pipelines/linux/sql-product-build-linux.yml [hunk 1]
```diff
@@ -131,8 +131,98 @@ steps:
       APP_ROOT=$(agent.builddirectory)/azuredatastudio-linux-x64
       APP_NAME=$(node -p "require(\"$APP_ROOT/resources/app/product.json\").applicationName")
       INTEGRATION_TEST_ELECTRON_PATH="$APP_ROOT/$APP_NAME" \
-      DISPLAY=:10 ./scripts/test-extensions-unit.sh
-    displayName: 'Run Stable Extension Unit Tests'
+      DISPLAY=:10 node ./scripts/test-extensions-unit.js admin-tool-ext-win
```
Comment (Charles-Gagnon) [inline build/azure-pipelines/linux/sql-product-build-linux.yml:134]: Not a huge fan of the duplication here - it looks like we might be able to loop over the script part and just pass in the test names which seems like a much cleaner solution https://mattvsts.github.io/2019/05/04/solving-the-looping-problem-in-Azure-DevOps-Pipelines/

Comment (aaomidi) [reply build/azure-pipelines/linux/sql-product-build-linux.yml:134]: Oh cool! Didn't know they had this!

File: build/azure-pipelines/linux/sql-product-build-linux.yml [hunk 2]
```diff
@@ -1,3 +1,6 @@
+parameters:
+  extensions: []
```
Comment (Charles-Gagnon) [inline build/azure-pipelines/linux/sql-product-build-linux.yml:2]: Suggest naming this something like extensionsToUnitTest to make it clear that this shouldn't include ALL extensions

**Comment by aaomidi** [comment]
Which one? We have so man now :D @Charles-Gagnon

**Comment by Charles-Gagnon** [comment]
But only one with the honor to have the actual name ci.yml 😉 

https://github.com/microsoft/azuredatastudio/blob/main/.github/workflows/ci.yml

**Comment by aaomidi** [comment]
Oh, since that's not ADO i'll handle that later.

**Pull request merged**